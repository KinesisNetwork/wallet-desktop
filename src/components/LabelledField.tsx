import * as copy from 'copy-to-clipboard'
import * as React from 'react'

export interface Props {
  label: string
  value: string
  icon?: string
  isLoading?: boolean
  isCompact?: boolean
  isClipped?: boolean
  addon?: React.ReactNode
  isCopyable?: boolean
}

const clippedStyle = (isClipped: boolean | undefined) => isClipped ? { maxWidth: '35ch' } : {}

export const LabelledField: React.SFC<Props> = (props) => (
  <div className='field'>
    <label className='label is-small'>{props.label}</label>
    <div className='field is-grouped'>
      <div className={`control is-expanded ${props.isLoading && 'is-loading'}`} style={clippedStyle(props.isClipped)}>
        {
          props.isCopyable &&
          <button
            className='button is-text is-paddingless'
            onClick={() => copy(props.value)}
            title='Click to copy'
          >
            {props.value}
          </button>
        }
        {
          !props.isCopyable &&
          <p className='input is-static'>{props.value}</p>
        }
      </div>
      {props.addon && <div className='control'>{props.addon}</div>}
    </div>
  </div>
)

LabelledField.displayName = 'LabelledField'

export const HorizontalLabelledField: React.SFC<Props> = (props) => (
  <div className={`field is-horizontal ${props.isCompact ? 'is-marginless' : ''}`}>
    <div className='field-label is-small'>
      <label className='label'>{props.label}</label>
    </div>
    <div className='field-body'>
      <div className='field is-grouped'>
        <div className={`control is-expanded ${props.isLoading && 'is-loading'}`} style={clippedStyle(props.isClipped)}>
          <p className='input is-static is-small is-block is-clipped' style={{ textOverflow: 'ellipsis' }} >
            {props.value}
          </p>
        </div>
        {props.addon && <div className='control'>{props.addon}</div>}
      </div>
    </div>
  </div>
)

HorizontalLabelledField.displayName = 'HorizontalLabelledField'
