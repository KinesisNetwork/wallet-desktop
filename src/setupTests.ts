import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
