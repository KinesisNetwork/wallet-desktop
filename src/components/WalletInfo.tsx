import * as React from 'react'

import { LabelledField } from '@components/LabelledField'
import { renderAmount } from '@services/util'

export interface Props {
  publicKey: string
  accountBalance: string
  accountName: string
  isAccountLoading: boolean
}

export class WalletInfo extends React.Component<Props> {
  displayName = 'WalletInfo'

  constructor(props: Props) {
    super(props)
  }

  render() {
    return (
      <React.Fragment>
        <h1 className='sub-heading primary-font'>Account Information</h1>
        <div className='tabs is-small'>
          <ul>
            <li className='is-active'>
              <a>
                <span className='icon is-small has-text-primary'><i className='fas fa-cubes' /></span>
                <span>KAU</span>
              </a>
            </li>
            <li>
              <a className='is-disabled' title='Coming soon!'>
                <span className='icon is-small has-text-grey-light'><i className='fas fa-cubes' /></span>
                <span>KAG</span>
              </a>
            </li>
            <li><a className='is-disabled' title='Coming soon!'>KWG</a></li>
            <li><a className='is-disabled' title='Coming soon!'>KWS</a></li>
          </ul>
        </div>
        <LabelledField label='Account Name' value={this.props.accountName} />
        <LabelledField
          label='Kinesis Balance'
          value={renderAmount(this.props.accountBalance)}
          isLoading={this.props.isAccountLoading}
        />
        <LabelledField label='Public Key' value={this.props.publicKey} isCopyable={true} />
        <div className='is-divider is-hidden-tablet' style={{ margin: 0, borderTopWidth: '0.01rem' }} />
      </React.Fragment>
    )
  }
}
