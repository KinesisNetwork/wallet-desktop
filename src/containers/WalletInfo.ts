import { connect } from 'react-redux'

import { WalletInfo as WalletInfoPresentation } from '@components/WalletInfo'
import { getActiveAccount } from '@selectors'
import { RootState } from '@store'

const mapStateToProps = (state: RootState) => {
  const activeAccount = getActiveAccount(state.wallet)
  return {
    accountName: activeAccount.name,
    publicKey: activeAccount.keypair.publicKey(),
    accountBalance: 0,
    isAccountLoading: false,
  }
}

export const WalletInfo = connect(mapStateToProps)(WalletInfoPresentation)
