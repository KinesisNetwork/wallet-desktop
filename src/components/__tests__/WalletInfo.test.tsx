import { shallow } from 'enzyme';
import * as React from 'react'

import { LabelledField } from '@components/LabelledField';
import { WalletInfo } from '@components/WalletInfo'
import '../../setupTests'

describe('WalletInfo', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<WalletInfo publicKey='' accountBalance='' accountName='' isAccountLoading={true} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should have a primary heading', () => {
    const wrapper = shallow(<WalletInfo publicKey='' accountBalance='' accountName='' isAccountLoading={true} />)

    const heading = wrapper.find('h1')

    expect(heading.text()).toEqual('Account Information')
  })

  it('renders four list items in the tab', () => {
    const wrapper = shallow(<WalletInfo publicKey='' accountBalance='' accountName='' isAccountLoading={true} />)

    const tabs = wrapper.find('li')

    expect(tabs).toHaveLength(4)
    expect(tabs.at(0).text()).toEqual('KAU')
    expect(tabs.at(1).text()).toEqual('KAG')
    expect(tabs.at(2).text()).toEqual('KWG')
    expect(tabs.at(3).text()).toEqual('KWS')
    expect(tabs.at(1).find('a').hasClass('is-disabled')).toBe(true)
    expect(tabs.at(2).find('a').hasClass('is-disabled')).toBe(true)
    expect(tabs.at(3).find('a').hasClass('is-disabled')).toBe(true)
  })

  it('renders three LabelledField components', () => {
    const wrapper = shallow(<WalletInfo publicKey='' accountBalance='' accountName='' isAccountLoading={true} />)

    expect(wrapper.find(LabelledField)).toHaveLength(3)
  })
})
