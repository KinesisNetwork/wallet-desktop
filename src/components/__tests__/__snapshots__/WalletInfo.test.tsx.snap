// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletInfo renders correctly 1`] = `
<Fragment>
  <h1
    className="sub-heading primary-font"
  >
    Account Information
  </h1>
  <div
    className="tabs is-small"
  >
    <ul>
      <li
        className="is-active"
      >
        <a>
          <span
            className="icon is-small has-text-primary"
          >
            <i
              className="fas fa-cubes"
            />
          </span>
          <span>
            KAU
          </span>
        </a>
      </li>
      <li>
        <a
          className="is-disabled"
          title="Coming soon!"
        >
          <span
            className="icon is-small has-text-grey-light"
          >
            <i
              className="fas fa-cubes"
            />
          </span>
          <span>
            KAG
          </span>
        </a>
      </li>
      <li>
        <a
          className="is-disabled"
          title="Coming soon!"
        >
          KWG
        </a>
      </li>
      <li>
        <a
          className="is-disabled"
          title="Coming soon!"
        >
          KWS
        </a>
      </li>
    </ul>
  </div>
  <Component
    label="Account Name"
    value=""
  />
  <Component
    isLoading={true}
    label="Kinesis Balance"
    value="0"
  />
  <Component
    isCopyable={true}
    label="Public Key"
    value=""
  />
  <div
    className="is-divider is-hidden-tablet"
    style={
      Object {
        "borderTopWidth": "0.01rem",
        "margin": 0,
      }
    }
  />
</Fragment>
`;
